composite_broker/src/main.rs

```diff
@@ -1,3 +1,706 @@
-fn main() {
-    println!("Hello, world!");
+mod broker;
```

Comment:
I don't think you're using rustfmt on each of these files, but it is great! Makes all of the code style match the normal expectations without having to worry about it yourself.